File: java/amp-eventing/src/main/java/amp/eventing/streaming/DefaultEventStreamFactory.java
package amp.eventing.streaming;

import cmf.eventing.patterns.streaming.IEventStream;

public class DefaultEventStreamFactory implements IEventStreamFactory {

    private IStandardStreamingEventBus eventBus;
    private String topic;

    @Override
    public void setEventBus(IStandardStreamingEventBus bus) {
        this.eventBus = bus;
    }

    @Override
    public void setTopic(String topic) {
        this.topic = topic;
    }

    @Override
    public IEventStream generateEventStream() {
        return new DefaultEventStream(eventBus, topic);
    }
}

File: dotnet/source/amp.eventing/streaming/DefaultEventStreamFactory.cs
using cmf.eventing.patterns.streaming;

namespace amp.eventing.streaming
{
    public class DefaultEventStreamFactory : IEventStreamFactory 
    {
        private IStandardStreamingEventBus _eventBus;
        private string _topic;

        public string Topic { set { _topic = value; } }

        public IEventStream GenerateEventStream()
        {
            return new DefaultEventStream(_eventBus, _topic);
        }

        public IStandardStreamingEventBus EventBus
        {
            set { _eventBus = value; }
        }
    }
}
